### PR TITLE
fix(graphql): send connection_init message during handshake

### PR DESCRIPTION
**Pull request by othorin**
Describe the purpose of the pull request.

#### Fixes / Enhancements
The implementation of `graphql-transport-ws` fails on the first message as it's only sent if the connection controller is in the `connected` state. For this protocol, the state is instead set to `handshake` during the first `_write()`.

https://github.com/zino-hofmann/graphql-flutter/blob/02be959597f0ffe0c21a227a2d1fdb02b3f56831/packages/graphql/lib/src/links/websocket_link/websocket_client.dart#L250-L256

Without this, the connection fails with the following log:
```
flutter: Initialising connection
flutter: There was an error causing connection lost: Bad state: No element
flutter: Disconnected from websocket.
```

### CODE REVIEW

File: melos.yaml [hunk 1]
```diff
@@ -65,7 +65,7 @@ scripts:
 
   client_test_coverage:
     description: Run tests in a specific package.
-    run: melos exec --concurrency=2 -- "dart run test --coverage="coverage" && dart run coverage:format_coverage --lcov --in=coverage --out=coverage.lcov --packages=.packages --report-on=lib"
+    run: melos exec --concurrency=2 -- "dart run test --coverage="coverage" && dart run coverage:format_coverage --lcov --in=coverage --out=coverage.lcov --packages=.dart_tool/package_config.json --report-on=lib"
```
Comment (vincenzopalazzo) [inline melos.yaml:68]: good catch

**Review by vincenzopalazzo — APPROVED** [review]
Very good Job, thank you so much to add some tests for this part of the code that was untested at all.

@budde377 any additional comment? for me is ready to land on master!

**Review by budde377 — APPROVED** [review]
Thanks for the tests and fix! LGTM @vincenzopalazzo

**Comment by othorin** [comment]
Thank you!

**Pull request merged**